database/events.go

```diff
@@ -48,41 +96,54 @@ func NewEventsDBModule(db CloudtrustDB) EventsDBModule {
 func (cm *eventsDBModule) Store(_ context.Context, m map[string]string) error {
 	// if ctEventType is not "", then record the events in MariaDB
 	// otherwise, do nothing
-	if m["ct_event_type"] == "" {
+	if m[CtEventType] == "" {
 		return nil
 	}
 
 	// the event was already formatted according to the DB structure already at the component level
 
 	//auditTime - time of the event
-	auditTime := m["audit_time"]
+	auditTime := m[CtEventAuditTime]
 	// origin - the component that initiated the event
-	origin := m["origin"]
+	origin := m[CtEventOrigin]
 	// realmName - realm name of the user that is impacted by the action
-	realmName := m["realm_name"]
+	realmName := m[CtEventRealmName]
 	//agentUserID - userId of who is performing an action
-	agentUserID := m["agent_user_id"]
+	agentUserID := m[CtEventAgentUserID]
 	//agentUsername - username of who is performing an action
-	agentUsername := m["agent_username"]
+	agentUsername := m[CtEventAgentUsername]
 	//agentRealmName - realm of who is performing an action
-	agentRealmName := m["agent_realm_name"]
+	agentRealmName := m[CtEventAgentRealmName]
 	//userID - ID of the user that is impacted by the action
-	userID := m["user_id"]
+	userID := m[CtEventUserID]
 	//username - username of the user that is impacted by the action
-	username := m["username"]
+	username := m[CtEventUsername]
 	// ctEventType that  is established before at the component level
-	ctEventType := m["ct_event_type"]
+	ctEventType := m[CtEventType]
 	// kcEventType corresponds to keycloak event type
-	kcEventType := m["kc_event_type"]
+	kcEventType := m[CtEventKcEventType]
 	// kcOperationType - operation type of the event that comes from Keycloak
-	kcOperationType := m["kc_operation_type"]
+	kcOperationType := m[CtEventKcOperationType]
 	// Id of the client
-	clientID := m["client_id"]
+	clientID := m[CtEventClientID]
 	//additional_info - all the rest of the information from the event
-	additionalInfo := m["additional_info"]
+	additionalInfo := m[CtEventAdditionalInfo]
+	if additionalInfo == "" {
+		var addNfo = make(map[string]string)
+		for k, v := range m {
+			if !isInArray(ctEventColumns, k) {
+				addNfo[k] = v
+			}
+		}
+		if additionalInfoBytes, err := json.Marshal(addNfo); err == nil && len(addNfo) > 0 {
+			additionalInfo = string(additionalInfoBytes)
+		}
+	}
 
 	//store the event in the DB
-	_, err := cm.db.Exec(insertEvent, auditTime, origin, realmName, agentUserID, agentUsername, agentRealmName, userID, username, ctEventType, kcEventType, kcOperationType, clientID, additionalInfo)
+	_, err := cm.db.Exec(insertEvent, auditTime, origin, checkNull(realmName), checkNull(agentUserID), checkNull(agentUsername),
```

Comment:
Why do we need to check for empty string ?

Reply:
It would be better to have NULL in database rather than empty string.